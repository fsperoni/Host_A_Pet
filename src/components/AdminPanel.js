import React from "react";
import Roles from "./Roles";

const AdminPanel = () => {

  return (
    <div className="AdminPanel">
      <div className="container col-md-6 offset-md-3 col-lg-4 offset-lg-4">
        <h2 className="mb-3">Admin Panel</h2>
        <Roles />
      </div>
    </div>
  );
}

export default AdminPanel;