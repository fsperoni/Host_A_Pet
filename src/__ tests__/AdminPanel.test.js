import { render } from "@testing-library/react";
import AdminPanel from "../components/AdminPanel";
import UserContext from "../hooks/useUserContext";

const {
  currentAdminUser
} = require("./_testCommon");

// smoke test
it('renders without crashing', function() {
  render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <AdminPanel />
    </UserContext.Provider>);
});

//snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <AdminPanel />
    </UserContext.Provider>
  );
  expect(asFragment()).toMatchSnapshot();
});

it("should have 'Admin Panel' in the component.", function() {
  const { getByText } = render(
    <UserContext.Provider
          value={{ currentUser: currentAdminUser() , function() {}}}>
      <AdminPanel />
    </UserContext.Provider>
  );
  expect(getByText("Admin Panel")).toBeInTheDocument();
});