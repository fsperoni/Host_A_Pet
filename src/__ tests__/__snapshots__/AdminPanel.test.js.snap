// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="AdminPanel"
  >
    <div
      class="container col-md-6 offset-md-3 col-lg-4 offset-lg-4"
    >
      <h2
        class="mb-3"
      >
        Admin Panel
      </h2>
      <div
        class="Roles"
      >
        <div
          class="container"
        >
          <h2
            class="mb-3"
          >
            Roles
          </h2>
          <h3
            class="ms-2"
          >
            No roles added yet
          </h3>
        </div>
        <div
          class="RoleAddForm mt-3 mb-3"
        >
          <div
            class="container"
          >
            <h2
              class="mb-3"
            >
              Add a Role
            </h2>
            <div
              class="card"
            >
              <div
                class="card-body"
              >
                <form>
                  <div
                    class="form-group"
                  >
                    <label>
                      Name:
                    </label>
                    <input
                      class="form-control"
                      name="name"
                      required=""
                      value=""
                    />
                  </div>
                  <div
                    class="container mt-4"
                  >
                    <button
                      class="btn btn-primary btn-block"
                    >
                      Add
                    </button>
                    <button
                      class="btn btn-warning btn-block ms-4"
                    >
                      Cancel
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
